pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -57,3 +58,36 @@ func AddCertKeyToContext(ctx context.Context, certificate, key string) (context.
 
 	return oidc.ClientContext(ctx, client), nil
 }
+
+func FetchAuthURL(config map[string]interface{}) (string, error) {
+	// If the authEndpoint is already configured, use that
+	if authURL, ok := config["authEndpoint"]; ok {
+		return authURL.(string), nil
+	}
+
+	issuerURL, ok := config["issuer"].(string)
+	if !ok {
+		return "", fmt.Errorf("both authEndpoint and issuerURL are missing in the authConfig")
+	}
+
+	discoveryURL := fmt.Sprintf("%s/.well-known/openid-configuration", issuerURL)
+	resp, err := http.Get(discoveryURL)
```

Comment:
I think we should avoid using the default http client. A custom one with a reasonable Timeout set would be better.